tasks/main.yml

```diff
@@ -1,36 +1,34 @@
-- name: Set database to mysql
-  debconf:
-    name: sympa
-    question: sympa/database-type
-    value: mysql
-    vtype: select
+---
+## Sympa configuration main tasks file
 
-- name: Set webserver to other
-  debconf:
-    name: sympa
-    question: wwsympa/webserver_type
-    value: Other
-    vtype: select
+- name: MySQL/MariaDB configuration
+  include_tasks: mysql.yml
+  when: sympa_db_type == 'mysql'
+
+- name: PostgreSQL configuration
+  include_tasks: postgresql.yml
+  when: sympa_db_type == 'PostgreSQL'
 
-- name: "Set dbconfig-install to no"
+- name: Set debconf options for sympa
   debconf:
-    name: sympa
-    question: sympa/dbconfig-install
-    value: no
-    vtype: boolean
-  changed_when: false
-
-- name: Create a new database with name '{{ sympa_db_name }}'
-  mysql_db:
-    name: "{{ sympa_db_name }}"
-    encoding: utf8
-
-- name: Create DB user '{{ sympa_db_user }}'
-  mysql_user:
-    name: "{{ sympa_db_user }}"
-    password: "{{ sympa_db_password | mandatory }}"
-    priv: '{{ sympa_db_name }}.*:ALL,GRANT'
-  no_log: True
+    name: "{{ item.name }}"
+    question: "{{ item.question }}"
+    value: "{{ item.value }}"
+    vtype: "{{ item.vtype }}"
+  loop:
+      ## set the database type
+  - { name: 'sympa', question: 'sympa/database-type', value: '{{ sympa_db_type }}' , vtype: select }
+      ## Configure the database at sympa installation with dbconfig
+  - { name: 'sympa', question: 'sympa/dbconfig-install', value: 'true' , vtype: boolean }
```

Comment:
do we want this setting? Arent we configuring the database on our own

Reply:
I tried again: for PostgreSQL, when I set this to `false`, sympa installation fails with the message: 
```
Database sympa defined in sympa.conf has not the right structure or is unreachable. verify db_xxx parameters in sympa.conf
```
(and I tried again with postgresql or PostgreSQL for sympa parameter db_type, and that's the same). 

For MySQL, when I tried yesterday there was no difference wether I put true or false. I'll try to figure out what is the difference between PostgreSQL and MySQL installation.

Reply:
I'm too puzzled with this dbconfig parameter. I think my [initial interpretation](https://github.com/UdelaRInterior/ansible-sympa/issues/3#issuecomment-590122189) was right: seeing `sympa.conf` file, sympa tries an upgrade but, with PostreSQL, fails when running: 
```
# /usr/lib/sympa/bin/sympa.pl --health_check
```
One strange thing is that I don't find any reference to `db_passwd` in [database drivers code](https://github.com/sympa-community/sympa/tree/sympa-6.2/src/lib/Sympa/DatabaseDriver). `dbconfig` must do something that I don't identify, that solves this issue. 

In order to avoid staying stucked, I propose you to set `sympa/dbconfig-install` to `true` for PostgreSQL and to maintain it set to `false` for MySQL.

Reply:
Possible solution [here](https://github.com/stuvusIT/sympa/pull/11#pullrequestreview-364626564). I'll test again.

Reply:
> Possible solution [here](https://github.com/stuvusIT/sympa/pull/11#pullrequestreview-364626564). I'll test again.

Eureka! Yes, it did!